cs/src/Contracts/TunnelAccessControlEntry.cs

```diff
@@ -141,6 +153,8 @@ public TunnelAccessControlEntry()
         /// <remarks>
         /// These must be one or more values from <see cref="TunnelAccessScopes" />.
         /// </remarks>
+        [MaxLength(MaxScopes)]
+        [ArrayStringLength(TunnelAccessScopes.MaxLength)]
```

Comment:
It would be great to have a regex validating known scopes and their derivatives (i.e. multiple scopes delimited by space)

Reply:
For that to be maintainable, the pattern needs to be generated from the list of all scopes. But `ValidateRegularExpressionAttribute` requires a compile-time string constant... unless I extend the attribute to take a method name or something and then use reflection, or use a source-generator, but either of those approaches would be messy. I think it's good enough to just do length validation here, and leave the rest of the validation to the code in the service.